app/controllers/api/v0/plans_controller.rb

```diff
@@ -61,10 +61,44 @@ def index
     if params[:per_page].present? && params[:per_page].to_i > Rails.configuration.branding[:application][:api_max_page_size]
       params[:per_page] = Rails.configuration.branding[:application][:api_max_page_size]
     end
-    @plans = paginate @user.org.plans.includes( [ {answers: :question_options} ,
-      template: [ { phases: { sections: { questions: :question_format } } }, :org]
+    @plans = paginate @user.org.plans.includes( [ {roles: :user}, {answers: :question_options} ,
+      template: [ { phases: { sections: { questions: [:question_format, :themes ] } } }, :org]
       ])
+    
+    # Filter on list of users
+    user_ids = extract_param_list(params, "user")
+    @plans = @plans.where(roles: {user_id: user_ids}) if user_ids.present?
+    # filter on dates
+    if params["created_after"].present? || params["created_before"].present?
+      @plans = @plans.where(created_at: dates_to_range(params,"created_after","created_before"))
+    end
+    if params["updated_after"].present? || params["updated_before"].present?
+      @plans = @plans.where(updated_at: dates_to_range(params,"updated_after","updated_before"))
+    end
+    # filter on funder (dmptemplate_id)
+    template_ids = extract_param_list(params, "template")
+    @plans = @plans.where(templates: {family_id: template_ids}) if template_ids.present?
+    # filter on id(s)
+    plan_ids = extract_param_list(params, "plan")
+    @plans = @plans.where(id: plan_ids ) if plan_ids.present?
     respond_with @plans
   end
 
+  private
+
+  def extract_param_list(params, attribute)
+    list = params.fetch(attribute+"[]", [])
+    val = params.fetch(attribute, [])
+    list << val if val.present?
+    list
+  end
+
+  # takes in the params hash and converts to a date-range
+  def dates_to_range(hash,start,stop)
+    today = Date.today
+    start_date = Date.parse(hash.fetch(start, today.prev_month.to_date.to_s))
```

Comment:
Would be useful to trap any exceptions thrown by the `Date.parse` here and either send the error back to the caller or default to a specific date like one month ago or something

Reply:
We should add that in and make it apply both here and in the statistics_controller, as I've copied this method over from there.